### DISCUSSION TITLE
I am stepping down from MkDocs

### DISCUSSION THREAD

**Discussion by oprypin** [mkdocs/mkdocs#3621] [Announcements]
3 years ago I was given push access to MkDocs among a few other people, without a single word regarding how anything should be done. Even the ability to make a release hinged on a leftover API token. At that time it became pretty clear that I'd be doing most of the work, but the whole time I was trying to operate in a way that would follow whatever my best guess was regarding the project's philosophy thus far. A year later I got admin access to GitHub and another year later to PyPI as well. Slowly over time I was getting a bit more confident in making changes and was able to steer the project in a slightly new direction. Also at a very late time, without any fanfare, I finally got confident to ask for donations for my work on MkDocs. (I'll expand on this part in a reply here.)

The entire time (that'd be for over 4 years now), me and @squidfunk were in disagreement about many things, but that was OK most of the time, he was doing his thing and I was doing my thing. I did explicitly ask him that I'd like to get more feedback regarding what MkDocs could be doing and to share ideas more. I also at one point inquired to increase our collaboration and contributed to mkdocs-material. But ultimately our conflicts led to @squidfunk unilaterally deciding to never again chat privately, only publicly. I don't think that was justified, but that's been the situation for a while now. Maintaining MkDocs has been really lonely, which, sure, for the most part is my fault, but still.

But the fact remained that we were all in the MkDocs organization on GitHub. And it remained baffling to me how I am supposed to keep a person in the organization that I can't even contact and who was never really doing anything for MkDocs itself anyway. One day this got to me, and without thinking too much, I dropped @squidfunk from the organization, so that it is clear that he does not in fact represent MkDocs. Upon which he decided to publicly complain about it - both on GitHub and as a passive-aggressive cryptic remark on Twitter.

Then all of a sudden, after 7 years of doing exactly nothing for MkDocs, @tomchristie stepped in without having any context, declared that there are "communication issues" (yeah, you bet...), removed me as admin, added @squidfunk back, and thus forced me to negotiate my status. @pawamoy created a chat room where I was thoroughly baffled how I'm supposed to resolve a conflict with @squidfunk in a semi-public setting if he's not willing to resolve anything in a private setting. I've been literally losing sleep over this for weeks, but ultimately I think there's nothing for me to say there, and nothing to be gained. 

Considering this, I am completely winding down my work on MkDocs, and probably also mkdocstrings and most other projects. I will still complete one last release of MkDocs along with the release notes, if that will be allowed.

This choice is not something I'd prefer to do, because being able to give my experience to the community was always the main highlight, and I will always miss it. But I will not take this level of disrespect.

At this time I would ask the admins to coordinate any ownership changes with me, and from my side I promise to likewise not make any changes. Luckily we do now have an actual communication channel with the admin, so that's convenient, I suppose.

If an alternate communication channel becomes needed, we can use https://github.com/oprypin/mkdocs/discussions/4, or the Gitter->Element chat is available.

**Comment by oprypin** [comment]
Here is an exhaustive list of money that I received in relation to MkDocs:

* $3000
  -assigned to myself out of Bloomberg's donation to MkDocs in August 2023

* $689.70
  -from donations to my GitHub Sponsors page that I started in July 2023 (where we can assume that 100% of the sponsors found me through MkDocs)
  -according to the "sponsorships export" as of today

(About half of this will be eaten by taxes.)

---

We can work out some announcement that active sponsors should switch away from me to some other recipient that MkDocs will decide. For now I am keeping this as is.

**Reply by oprypin** [reply]
I have now exited the "mkdocs" organization on GitHub.

I have sent this message as an "email update" to sponsors on GitHub:

> ### @oprypin stopping work on MkDocs
>
> Hi!
>
> I have now stopped working on MkDocs as other individuals have taken over it. [This was my post about it.](https://github.com/mkdocs/mkdocs/discussions/3621)
>
> I assume that most of my sponsors joined in order to support current and future work on [MkDocs](https://github.com/mkdocs/mkdocs). If that's the case for you, **please cancel the [sponsorship](https://github.com/sponsors/oprypin)** of me, because I don't plan to work on MkDocs at all anymore.
>
> I did publish one last release though - check it out! - [MkDocs 1.6.0](https://github.com/mkdocs/mkdocs/releases/tag/1.6.0)
>
> In any case, thanks a lot for your support!

Thanks to the team for letting me stay in the meantime.

**Comment by squidfunk** [comment]
First of all, thanks for all your hard work on MkDocs. Again, I feel I need to stress that I always have admired and supported your work. However, I feel that your representation of things is a little skewed towards giving me the blame. Also, I'm not sure I understand why kicking me out of the MkDocs organization was necessary when [Tom was the one who added us to the team](https://github.com/mkdocs/mkdocs/discussions/2442#discussioncomment-964629). To the interested reader, [here's how the last 4 years played out from my point of view](https://github.com/mkdocs/mkdocs/issues/3579#issuecomment-1970691690). I've tried to resolve the situation countless times, but sadly to no avail.

Additionally, saying that I have done nothing for MkDocs is not quite right. Albeit I'm not a core team member, I was tagged many times, took part in countless conversations, gave feedback from a theme and plugin author perspective, helped out to solve issues and proposed [several PRs](https://github.com/mkdocs/mkdocs/pulls?q=is%3Apr+author%3Asquidfunk). As one can imagine, my time is pretty much tied to Material for MkDocs, a project that has become (almost) as popular as MkDocs itself, but I try my very best to help out in the entire ecosystem whenever I get tagged.

That being said, it's really sad to see you leave. Your input and work on this project was insanely valuable and despite our differences I always tried to be respectful. I respect your decision and wish you all the best for your future projects and endeavors.

---

PS: Yes, I blocked you on Gitter for the reasons mentioned in the linked comment, but my email address is located in every file that I ever released on GitHub. If you want to reach me privately, even now, it has always been and will always be possible.

**Comment by facelessuser** [comment]
It is unfortunate that leaving is the direction that is being taken, but I understand people need to do what they feel is best for them.

Regardless, thank you for all the work you've put into MkDocs. The work you've done has been extremely valuable to the community. You will be missed.

**Comment by Andre601** [comment]
Sad to see you go... And even sadder to see the context this is based on.

My biggest fear now is, that MkDocs will receive a negative hit in terms of contributions and updates, causing the situation it had several years ago before maintainership was moved.

I would've loved to help and contribute myself, but I lack the experience with python to make code contributions, so at most would've my additions be documentation changes, which I saw barely any reasons for...

Either way, I hope you can recover some of your strength back oprypin. And I would really love to see you around at places here on GitHub in the future, even if it was just for comments or alike. Take care!

**Reply by Andre601** [reply]
On an unrelated note, what methods are there to contact you? I don't use (nor really like tbh) Gitter. Tho if that is the only way, then I'll probs use it to chat with you, as I have a few questions I would like to ask you about privately. Only if you're up for it of course.

**Reply by oprypin** [reply]
@Andre601  Well it's not Gitter anymore anyway, it's Element, not too horrible of an application to use temporarily :) Or reach me by the email address from my commit messages (at which point don't be alarmed that I reply from a different email address)

**Reply by Andre601** [reply]
Yeah. I do use element for matrix, do only a bit sparely... Under what name can I find you there (Also, first need to figure out how to add a user there...)

**Comment by TheMikeyRoss** [comment]


**Reply by lovelydinosaur** [reply]
Oh do go away.

**Reply by TheMikeyRoss** [reply]
sorry

**Reply by lovelydinosaur** [reply]
No worries.

**Comment by lovelydinosaur** [comment]
That's unexpected - sorry to hear that @oprypin.

> This choice is not something I'd prefer to do, because being able to give my experience to the community was always the main highlight, and I will always miss it.

Okay. If you'd like to take some time or talk any of this through privately you're very welcome to do so.

> At this time I would ask the admins to coordinate any ownership changes with me

Yep no problem.

**Reply by oprypin** [reply]
For the record:
Since then, you further removed me from ownership of the repository itself and blocked merging my own commits.
I'll be glad to lose the rights as soon as the release is out, but for now this is further insulting.
These are currently blocked https://github.com/mkdocs/mkdocs/milestone/18

**Reply by lovelydinosaur** [reply]
Thanks Oleh... yes, I've enabled protected branch settings on the repository.
It's just setting things up with some more sensible defaults in place.

**Comment by pawamoy** [comment]
Oh well, I did not expect that. I hope you'll be able to recover from this and keep doing what you love, even if not in MkDocs/mkdocstrings. Thanks for your contributions to this community. I'll welcome you if you ever want to come work with me again in the future.

**Comment by ultrabug** [comment]
Dear @oprypin , I can't imagine how painful this decision was to make for you and I'm sorry you suffered personally about this situation. You made your talents available to this project and brought it far further than anyone could have wished for.

In retrospect, I also have the feeling that this project allowed you to overcome some barriers of your own while at the same time it was hard not to feel alone. I do take my blame on this as not being able to dedicate more time to the project and you as a person.

I hope and wish you'll recover from this quickly and find joy somewhere your interests will guide you to.

Thank you for your amazing work and for the rigor you handled things like grants/sponsoring.

_To the maybe future reader of this thread: this is a hard reminder that Tech is made by people who have not only technical opinions but feelings as well. Balancing both is a daunting task that we should always try to reach for the good of both a project and its people._

**Comment by Guts** [comment]
Hey @oprypin;

I'm arriving a little late here and I'm not sure I have all the elements to understand the situation, but I think it's a real shame.

From my point of view as a plugin developer, the development of Mkdocs has been on a roll since you took over.

It's a real waste that personal problems are preventing all the good will and skills from feeding this beautiful ecosystem around the project, but that's what open source is all about I guess: people. With what works and what doesn't.

**Comment by ghost** [comment]
Hey @oprypin  

Thanks for the congrats on MKdocs, I really like mkdocs and have made it my personal blog!
